server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/DLPRulesRoutes.java

```diff
@@ -0,0 +1,206 @@
+
+            dlpConfigurationStore.store(senderDomain, dlpConfigurations);
+
+            response.status(HttpStatus.NO_CONTENT_204);
+            return EMPTY_BODY;
+        });
+    }
+
+    @GET
+    @Path("/{senderDomain}")
+    @ApiOperation(value = "Retrieve a list of dlp configs for given senderDomain")
+    @ApiImplicitParams({
+        @ApiImplicitParam(required = true, dataType = "string", name = "senderDomain", paramType = "path")
+    })
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.OK_200, message = "OK. dlp configs returned"),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Invalid senderDomain in request"),
+        @ApiResponse(code = HttpStatus.NOT_FOUND_404, message = "The domain does not exist."),
+        @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500,
+            message = "Internal server error - Something went bad on the server side.")
+    })
+    public void defineList() {
+        service.get(SPECIFIC_DLP_RULE_DOMAIN, (request, response) -> {
+            Domain senderDomain = checkValidDomain(request);
+            List<DLPConfigurationItem> dlpConfigurations = dlpConfigurationStore
+                .list(senderDomain)
+                .collect(Guavate.toImmutableList());
+
+            List<DLPConfigurationDTO> dtOs = toDTOs(dlpConfigurations);
+            response.status(HttpStatus.OK_200);
+            response.header(CONTENT_TYPE, JSON_CONTENT_TYPE);
+            return dtOs;
+        }, jsonTransformer);
+    }
+
+    @DELETE
+    @Path("/{senderDomain}")
+    @ApiOperation(value = "Clear all dlp configs for given senderDomain")
+    @ApiImplicitParams({
+        @ApiImplicitParam(required = true, dataType = "string", name = "senderDomain", paramType = "path")
+    })
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.OK_200, message = "OK. dlp configs are cleared"),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Invalid senderDomain in request"),
+        @ApiResponse(code = HttpStatus.NOT_FOUND_404, message = "The domain does not exist."),
+        @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500,
+            message = "Internal server error - Something went bad on the server side.")
+    })
+    public void defineClear() {
+        service.delete(SPECIFIC_DLP_RULE_DOMAIN, (request, response) -> {
+            Domain senderDomain = checkValidDomain(request);
+            dlpConfigurationStore.clear(senderDomain);
+
+            response.status(HttpStatus.NO_CONTENT_204);
+            return EMPTY_BODY;
+        }, jsonTransformer);
+    }
+
+    private Domain checkValidDomain(Request request) {
+        String domainName = request.params(DOMAIN_NAME);
+        try {
+            Domain domain = Domain.of(domainName);
+            checkDomainInList(domain);
+
+            return domain;
+        } catch (IllegalArgumentException | DomainListException e) {
```

Comment:
Here a DomainListException means 500, no?

Reply:
I liked what Antoine did in previous PR, IE extracting ErrorResponder calls into separated methods, like `invalidDomain()` `invalidRequest`() `notFound()` `serverError()`...

Could we extract this?